src/OpenAL/OpenToolkit.OpenAL/Native/ApiBase.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Text;
+using OpenToolkit.Core.Loader;
+
+namespace OpenToolkit.Audio.OpenAL
+{
+    public abstract class ApiBase
+    {
+        private static Dictionary<string, IPlatformLibraryNameContainer> DllIntercepts = new Dictionary<string, IPlatformLibraryNameContainer>();
+
+        static ApiBase()
+        {
+            NativeLibrary.SetDllImportResolver(typeof(ApiBase).Assembly, ImportResolver);
+        }
+
+        protected static IntPtr ImportResolver(string libraryName, Assembly assembly, DllImportSearchPath? searchPath)
+        {
+            IntPtr libHandle = IntPtr.Zero;
+            if (DllIntercepts.TryGetValue(libraryName, out var libraryNameContainer))
+            {
+                string libName = libraryNameContainer.GetLibraryName();
+
+                // FIXME: Figure out the search path stuf
+                NativeLibrary.TryLoad(libName, assembly, searchPath, out libHandle);
```

Comment:
Just saw that one now. `TryLoad` suggests it returns a bool, whether it succeeds or not. I think this return value should also be handled?

Reply:
Yes, but I'm not sure what to do if it fails. Exception?

Reply:
Id say yes. maybe there is a `PlatformException` type already defined somewhere in System.* which you could throw

Reply:
Fixed in latest